src/api/app/models/worker_status.rb

```diff
@@ -44,88 +44,91 @@ def project_names(prjs)
 
   def update_workerstatus_cache
     # do not add hiding in here - this is purely for statistics
-    ret = Backend::Api::BuildResults::Worker.status
-    wdata = Xmlhash.parse(ret)
+    backend_status = Backend::Api::BuildResults::Worker.status
+    wdata = Nokogiri::XML(backend_status)
     @mytime = Time.now.to_i
-    @squeues = {}
-    Rails.cache.write('workerstatus', ret, expires_in: 3.minutes)
+    @squeues = Hash.new(0)
+    Rails.cache.write('workerstatus', backend_status, expires_in: 3.minutes)
     StatusHistory.transaction do
-      wdata.elements('blocked') do |e|
+      wdata.search('//blocked').each do |e|
         save_value_line(e, 'blocked')
       end
-      wdata.elements('waiting') do |e|
+      wdata.search('//waiting').each do |e|
         save_value_line(e, 'waiting')
       end
-      wdata.elements('partition') do |p|
-        p.elements('daemon') do |daemon|
-          parse_daemon_infos(daemon)
-        end
+      wdata.search('partition/daemon').each do |daemon|
+        parse_daemon_infos(daemon)
       end
       parse_worker_infos(wdata)
       @squeues.each_pair do |key, value|
         StatusHistory.create(time: @mytime, key: key, value: value)
       end
     end
-    ret
+    backend_status
   end
 
   private
 
   def add_squeue(key, value)
-    @squeues[key] ||= 0
     @squeues[key] += value.to_i
   end
 
   def parse_daemon_infos(daemon)
-    return unless daemon['type'] == 'scheduler'
-    arch = daemon['arch']
+    return unless daemon.attributes['type'].value == 'scheduler'
     # FIXME2.5: The current architecture model is a gross hack, not connected at all
     #           to the backend config.
-    a = Architecture.find_by_name(arch)
-    if a
-      a.available = true
-      a.save
-    end
-    queue = daemon.get('queue')
+    queue = daemon.at_xpath('queue')
     return unless queue
```

Comment:
Can that happen that this XML https://build.opensuse.org/apidocs/workerstatus.xml in the `partition > daemon` section comes without queue?  @adrianschroeter ?